src/main/java/com/pgs/booking/service/WeatherService.java

```diff
@@ -0,0 +1,60 @@
+package com.pgs.booking.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pgs.booking.exception.ResourceNotFoundException;
+import com.pgs.booking.model.dto.WeatherDto;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Getter
+@Setter
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WeatherService {
+    @Value("${openwaether.api-key}")
+    private String keyValue;
+    private static final String mode = "json";
+    private static final String units = "imperial";
+    private final ObjectMapper objectMapper;
+    @Autowired
```

Comment:
do you need that at is autowired via constructor

Reply:
I have the final field with one constructor, so I do not need it